### ISSUE TITLE
New Feature to add and get Indexed Properties

### ISSUE THREAD

**Issue by ravenboilinux** [SharePoint/PnP-JS-Core#455]
#### Category
[ X] Enhancement

[ ] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [  2.0.5       ]

#### Expected / Desired Behavior / Question
Index Properties are just properties, in the SPWeb/SPFolder property-bag, that have been added to the key has been added to the vti_indexedpropertykeys property in the site. I wanted to know if we can add something like this

pnp.sp.web.indexedproperties.get() -- which would return all of the keys that are indexed

pnp.sp.web.indexedproperties.add(<property key name>)--which would first check if the property exists on the property bag and then base64 encodes and updates the vti_indexedpropertykeys with it

**Comment by patrick-rodgers** [comment]
I don't believe we can get at the property bags with REST, but has been a bit since I checked so will have another look.

**Comment by russgove** [comment]
i'm not sure of the context of the conversation but https://github.com/SharePoint/sp-dev-fx-webparts/tree/master/samples/react-property-bag-editor enables yo to edit the properrises of a site, and enables search based navigation using those proprties.

if we want to wrap. the property bag, we shoul also make sure we include the option to make properties searchable,.

👍

**Comment by patrick-rodgers** [comment]
Thanks @russgove - just had a look at that and the issue is that it users the sp.js library to do the actual updates which is a dependency we want to avoid. I do agree it would be great to wrap the props and be able to update them (including searchability). So will leave this one open but marked as awaiting API support for now.

**Comment by ravenboilinux** [comment]
The worst part is if you look at the SPPropertyValuesServerStub it has a method called SetFieldValue that is targeted to all client library targets
